docarray/index/abstract.py

```diff
@@ -793,15 +809,28 @@ def _validate_docs(
 
         return DocumentArray[BaseDocument].construct(out_docs)
 
-    def _to_numpy(self, val: Any) -> Any:
+    def _to_numpy(self, val: Any, allow_passthrough=False) -> Any:
+        """
+        Converts a value to a numpy array, if possible.
+
+        :param val: The value to convert
+        :param allow_passthrough: If True, the value is returned as-is if it is not convertible to a numpy array.
+            If False, a `ValueError` is raised if the value is not convertible to a numpy array.
+        :return: The value as a numpy array, or as-is if `allow_passthrough` is True and the value is not convertible
+        """
         if isinstance(val, np.ndarray):
             return val
-        elif isinstance(val, (list, tuple)):
+        if is_tf_available() and isinstance(val, TensorFlowTensor):
+            return val.unwrap().numpy()
+        if isinstance(val, (list, tuple)):
             return np.array(val)
-        elif torch_imported and isinstance(val, torch.Tensor):
+        if (torch_imported and isinstance(val, torch.Tensor)) or (
+            is_tf_available() and isinstance(val, tf.Tensor)
+        ):
             return val.numpy()
-        else:
-            raise ValueError(f'Unsupported input type for {type(self)}: {type(val)}')
+        if allow_passthrough:
+            return val
+        raise ValueError(f'Unsupported input type for {type(self)}: {type(val)}')
 
     def _convert_dict_to_doc(
         self, doc_dict: Dict[str, Any], schema: Type[BaseDocument]
```

Comment:
Can u also change `t_ = schema._get_field_type(field_name)` in this function to:
`t_ = unwrap_optional_type(schema._get_field_type(field_name))`

This will fix the error of row-wise database like Elasticsearch when handling Optional field like `ImageUrl`.